presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -248,57 +252,66 @@ private static ConnectorTableMetadata getTableMetadata(String schemaName, TpchTa
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint<ColumnHandle> constraint)
     {
-        TpchTableHandle table = (TpchTableHandle) tableHandle;
-        return new TableStatistics(new Estimate(getRowCount(table, Optional.of(constraint.getSummary()))), ImmutableMap.of());
+        TpchTableHandle tpchTableHandle = (TpchTableHandle) tableHandle;
+        String tableName = tpchTableHandle.getTableName();
+        TpchTable<?> tpchTable = TpchTable.getTable(tableName);
+
+        Predicate<Map<TpchColumn<?>, Object>> predicate = values -> {
+            TupleDomain<ColumnHandle> summary = constraint.getSummary();
+            Optional<Map.Entry<ColumnHandle, Domain>> unsupportedConstraint = summary.getDomains()
+                    .flatMap(domains -> domains.entrySet().stream()
+                            .filter(x -> !x.getKey().equals(toColumnHandle(OrderColumn.ORDER_STATUS)))
+                            .findAny());
+            if (unsupportedConstraint.isPresent()) {
+                return true;
```

Comment:
stats should match table layout. Tpch connector returns table layout for ORDER_STATUS even if there are constraints on other columns. You should keep such sementics